utility/MRAA/compatibility.cpp

```diff
@@ -10,33 +9,30 @@
  */
 void __msleep(int milisec)
 {
-    struct timespec req = {0};
-    req.tv_sec = 0;
-    req.tv_nsec = milisec * 1000000L;
-    nanosleep(&req, (struct timespec*)NULL);
-    //usleep(milisec*1000);
+    struct timespec req; // = {0};
+    req.tv_sec = (time_t)milisec / 1000;
+    req.tv_nsec = (milisec % 1000) * 1000000L;
+    clock_nanosleep(CLOCK_REALTIME, 0, &req, NULL);
 }
 
-void __usleep(int milisec)
+void __usleep(int microsec)
 {
-    struct timespec req = {0};
-    req.tv_sec = 0;
-    req.tv_nsec = milisec * 1000L;
-    nanosleep(&req, (struct timespec*)NULL);
-    //usleep(milisec);
+    struct timespec req; // = {0};
+    req.tv_sec = (time_t)microsec / 1000000;
+    req.tv_nsec = (microsec / 1000000) * 1000;
```

Comment:
should probably update this to
```c++
    req.tv_nsec = (microsec % 1000000) * 1000;
```

Reply:
Testing on RPi5. Modified for SPIDEV and MRAA in latest commit

Reply:
Hmm not sure if I need a new PR now or how to add to this one...